plonky2x/src/frontend/eth/beacon/builder.rs

```diff
@@ -40,7 +45,36 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         self.add_simple_generator(&generator);
         generator.validator
     }
+
+    /// SSZ Merkle Proof 
+    pub fn ssz_restore_merkle_root(
+        &mut self,
+        leaf: Bytes32Variable,
+        index: u64,
+        branch: Vec<Bytes32Variable>,
+    ) -> Bytes32Variable {
+        assert!(2u64.pow(branch.len() as u32 + 1) > index);
+        let mut hasher = leaf; // Initialize the hasher with the leaf node
+        for i in 0..branch.len() { // Iterate over each level of the Merkle tree
+            let (first, second) = if (index >> i) & 1 == 1 { 
```

Comment:
nit:  comments are overly used, can remove all of them. the code can speak for itself